src/components/common/ui/form/dropdown-mui/Dropdown.tsx

```diff
@@ -0,0 +1,143 @@
+/* eslint-disable react/prop-types */
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { ChevronDownIcon } from '@heroicons/react/24/solid';
+import { Box, Popper, Typography } from '@mui/material';
+import Autocomplete from '@mui/material/Autocomplete';
+import TextField from '@mui/material/TextField';
+import { useField } from 'formik';
+
+import { FieldState } from '@/components/common/ui/form/common/types';
+
+import { dropdown, input, popper, remark } from './Dropdown.styles';
+
+interface DropDownOption {
+  label: string;
+  id: string;
+}
+interface DropdownProps {
+  options: DropDownOption[];
+  label?: string;
+  name?: string;
+  isDisabled?: boolean;
+  placeholder?: string;
+  numberOfOptions?: number;
+  isSuccessOnDefault?: boolean;
+  defaultRemark?: string;
+  showRemark?: boolean;
+  size?: `${DropDownSize}`;
+  noOptionsText?: string;
+  onChange?: () => void;
+}
+
+export enum DropDownSize {
+  SMALL = 'small',
+  MEDIUM = 'medium',
+  LARGE = 'large',
+}
+
+export const Dropdown: React.FC<DropdownProps> = ({
+  options,
+  label = 'АНУ ТИЦЬНУВ',
+  placeholder = 'МОЛОДЧИНА! Тепер піди поспи солдат',
+  noOptionsText = 'Опції відсутні',
+  numberOfOptions = 4,
+  isSuccessOnDefault = false,
+  defaultRemark,
+  showRemark,
+  size = DropDownSize.MEDIUM,
+  isDisabled = false,
+  name,
+  onChange,
+}) => {
+  const inpRef = useRef<HTMLElement | null>(null);
+  const [isFocused, setIsFocused] = useState<boolean>(false);
+  const [inputValue, setInputValue] = useState('');
+  const popperRef = useRef<HTMLElement | null>(null);
+  const [filteredOptions, setFilteredOptions] = useState<number>(
+    options!.length,
+  );
+
+  const [{}, { touched, error }, { setTouched, setValue }] = useField(name);
+
+  let state = FieldState.DEFAULT;
```

Comment:
what is that? you can replace it with useMemo if needed. and rename variable, it is unclear name